huaweicloud/services/acceptance/acceptance.go

```diff
@@ -259,34 +254,36 @@ func (rc *ResourceCheck) CheckResourceDestroy() resource.TestCheckFunc {
 func (rc *ResourceCheck) checkResourceExists(s *terraform.State) error {
 	rs, ok := s.RootModule().Resources[rc.resourceName]
 	if !ok {
-		return fmtp.Errorf("Can not found the resource or data source in state: %s", rc.resourceName)
+		return fmt.Errorf("can not found the resource or data source in state: %s", rc.resourceName)
 	}
+
 	if rs.Primary.ID == "" {
-		return fmtp.Errorf("No id set for the resource or data source: %s", rc.resourceName)
+		return fmt.Errorf("No id set for the resource or data source: %s", rc.resourceName)
 	}
 	if strings.EqualFold(rc.resourceType, dataSourceTypeCode) {
 		return nil
 	}
 
-	if rc.getResourceFunc != nil {
-		conf := TestAccProvider.Meta().(*config.Config)
-		r, err := rc.getResourceFunc(conf, rs)
+	if rc.getResourceFunc == nil {
+		return fmt.Errorf("the 'getResourceFunc' is nil, please set it during initialization")
+	}
+
+	conf := TestAccProvider.Meta().(*config.Config)
+	r, err := rc.getResourceFunc(conf, rs)
+	if err != nil {
+		return fmt.Errorf("checking resource %s %s exists error: %s ",
+			rc.resourceName, rs.Primary.ID, err)
+	}
+
+	if rc.resourceObject != nil {
+		b, err := json.Marshal(r)
 		if err != nil {
-			return fmtp.Errorf("checking resource %s %s exists error: %s ",
+			return fmt.Errorf("marshaling resource %s %s error: %s ",
 				rc.resourceName, rs.Primary.ID, err)
 		}
-		if rc.resourceObject != nil {
-			b, err := json.Marshal(r)
-			if err != nil {
-				return fmtp.Errorf("marshaling resource %s %s error: %s ",
-					rc.resourceName, rs.Primary.ID, err)
-			}
-			json.Unmarshal(b, rc.resourceObject)
-		} else {
-			logp.Printf("[WARN] The 'resourceObject' is nil, please set it during initialization.")
-		}
+		json.Unmarshal(b, rc.resourceObject)
 	} else {
-		return fmtp.Errorf("The 'getResourceFunc' is nil, please set it.")
+		log.Printf("[WARN] The 'resourceObject' is nil, please set it during initialization")
```

Comment:
suggest move to up, keep same with CheckResourceDestroy

Reply:
I think it's unnecessary. if rc.resourceObject is nil, we just add a log rather than return a error.